Rework the Toolkit and TimescaleDB installation guides (#4336)
timescale/docs: self-hosted/tooling/install-toolkit.md

If you're using [$CLOUD_LONG][cloud], the $TOOLKIT_LONG is already installed.

> 

Reviewer: Can you add a prerequisite saying TimescaleDB should be installed with a link to the guides please.

Author: Two things: 
- Toolkit comes bundled in the HA image, meaning you can't/shouldn't have TimescaleDB installed as a prereq.
- The other tabs rely on different package managers, meaning the steps to install it are also slightly different.

Reviewer: Ok, let's do the whole page as tabs and for docker say you install with the ha so you don't need to pre-install. 

for the upgrage, point to the upgrade docker image.

## Install TimescaleDB Toolkit on self-hosted TimescaleDB

If you're hosting the $TIMESCALE_DB extension on your self-hosted database, you can install $TOOLKIT_SHORT by:

*   Using the TimescaleDB high-availability Docker image
*   Using a package manager such as `yum`, `apt`, or `brew` on platforms where
    pre-built binaries are available
*   Building from source. For more information, see the [$TOOLKIT_SHORT developer documentation][toolkit-gh-docs]

> <Tabs label="Install TimescaleDB Toolkit">

Reviewer: Can you add persistKey="os" to both tabs please, and labels that match the ones in the install guides please.

<Tab title="Docker">

> The recommended way to install the $TOOLKIT_SHORT is to use the

Reviewer: ```suggestion
Best practice for $TOOLKIT_SHORT installation is to use the
```

docker pull timescale/timescaledb-ha:pg17
```

> For more information on running $TIMESCALE_DB using Docker, see the section on

Reviewer: ```suggestion
For more information on running $TIMESCALE_DB using Docker, see 
```

> [pre-built containers][docker-install].

Reviewer: ```suggestion
[Install TimescaleDB from a Docker container][docker-install].
```

    
</Tab>

<Tab title="Red Hat-based" >

<Procedure>

> 1.  Make sure you have installed $TIMESCALE_DB and created a $TIMESCALE_DB

Reviewer: If you add the install guide to the prerequisites, you can remove this.

    ```

</Procedure>
    
</Tab>

<Tab title="Debian-based">

These instructions use the `apt` package manager. They have been tested on Ubuntu 20.04
and may also work on other Debian-based systems.

<Procedure>

> 1.  Make sure you have installed TimescaleDB and added the TimescaleDB

Reviewer: Same, can be removed if the install is a prerequisite

    ```

</Procedure>
    
</Tab>

</Tabs>

## Update Toolkit on self-hosted TimescaleDB

Update $TOOLKIT_SHORT by installing the latest version and running `ALTER EXTENSION`.

> <Tabs label="Tabs label for accessibility" >

Reviewer: persistKey etc.

<Tab title="Red Hat-based" >

<Procedure>

1.  Update your local repository list:

    ```bash
    yum update
    ```

1. Install the latest version of $TOOLKIT_LONG:

    ```bash
    yum install timescaledb-toolkit-postgresql-17
    ```

> 1.  Connect to the database where you want to use the new version of $TOOLKIT_SHORT.

Reviewer: Link to the connect page. Can you do that everywhere we say connect please.
